pkg/discovery/mdns.go

```diff
@@ -20,17 +20,34 @@ import (
 var log = logger.NewLogger("dapr.runtime.discovery")
 
 // RegisterMDNS uses mdns to publish an entry of the service to a local network
-func RegisterMDNS(id string, port int) error {
+func RegisterMDNS(id string, internalGrpcPort int, grpcPort int, httpPort int, metricsPort uint64, metricsEnabled bool, profilingPort int, profilingEnabled bool) error {
```

Comment:
is there a reason metrics port is `uint64` but http and grpc ports just `int`?

Reply:
Just because the component that parses the metrics CLI options currently stores the port as a `uint64` (for whatever reason).  That stored type could be changed (if it wasn't done for a specific reason), but ought to be a separate PR as its generally unrelated to this PR.